### PR TITLE
[screen-orientation][ios] Fix crash when reading rootViewController value

### PR DESCRIPTION
**Pull request by gabrieldonadel**
# Why

Closes ENG-9051

# How

This PR replaces the screen-orientation `UIApplication.shared.keyWindow` call with `UIApplication.shared.connectedScenes` as 'keyWindow' was deprecated in iOS 13.0 and it returns a key window across all connected scenes, which makes the app crash when a new window is displayed, E.g. the StoreReview popup 

# Test Plan

Run Expo Go Unversioned locally and test StoreReview + ScrenOrientation on NCL

https://github.com/expo/expo/assets/11707729/17ee2704-79f9-409e-a46c-6a397df46383

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-screen-orientation/CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,8 @@
 
 ### 🐛 Bug fixes
 
+- [ios] Fix crash when reading rootViewController value. ([#23039](https://github.com/expo/expo/pull/23039) by [@gabrieldonadel](https://github.com/gabrieldonadel))
```
Comment (tsapeta) [inline packages/expo-screen-orientation/CHANGELOG.md:11]:
```suggestion
- [iOS] Fix crash when reading `rootViewController` value. ([#23039](https://github.com/expo/expo/pull/23039) by [@gabrieldonadel](https://github.com/gabrieldonadel))
```

**Review by tsapeta — COMMENTED** [review]
Could you include a crash details before this change? I'm not actually sure why it fixes the issue 😅
Is the store review popup a separate window or scene? I think in most scenarios the `UIApplication.shared.keyWindow` is what we need, and now, with this change, we're basically doing the same thing but we pick the last (why?) key window 🤔

**Review by brentvatne — APPROVED** [review]

**Review by behenate — APPROVED** [review]
I did some testing and it seems to be working well in both Bare Expo and Go. Thanks for spotting and  fixing the issue!

File: packages/expo-screen-orientation/CHANGELOG.md (hunk 1 above)
Comment (behenate) [reply packages/expo-screen-orientation/CHANGELOG.md:11]: @tsapeta New code returns `EXRootViewController` and then `nil` while the popup is closing, in that case `screen-orientation` considers the supported interface orientations for that `ViewController` as an empty array and handles everything correctly.
While closing the popup `UIApplication.shared.keyWindow?.rootViewController` returns `UIApplicationRotationFollowingController` instead od the nil, it seems like `screen-orientation` has issues handling this. The issue looks like it's thread related . In all cases other than closing the popup output of the two functions seems to be te same

Comment (gabrieldonadel) [reply packages/expo-screen-orientation/CHANGELOG.md:11]: Yeah, it's exactly what Wojciech said

**Pull request merged**